src/components/MessageInput/hooks/messageInput.js

```diff
@@ -354,7 +354,10 @@ export default function useMessageInput(props) {
       type: 'setEmojiPickerIsOpen',
       value: true,
     });
-    document.addEventListener('click', closeEmojiPicker, false);
+
+    setTimeout(() =>
```

Comment:
Can we add a comment here as to why we need timeout?

Reply:
good call, added!

Reply:
How about useEffect?

```js
useEffect(() => {
    if (emojiPickerIsOpen) {
        document.addEventListener('click', closeEmojiPicker, false);
    }

    return () => {
        document.removeEventListener('click', closeEmojiPicker, false);
    }
}, [emojiPickerIsOpen])
```